integration-cli/docker_cli_daemon_test.go

```diff
@@ -403,3 +403,77 @@ func TestDaemonKeyMigration(t *testing.T) {
 
 	logDone("daemon - key migration")
 }
+
+// Simulate an older daemon (pre 1.3) coming up with volumes specified in containers
+//	without corrosponding volume json
+func TestDaemonUpgradeWithVolumes(t *testing.T) {
+	d := NewDaemon(t)
+
+	graphDir := filepath.Join(os.TempDir(), "docker-test")
+	defer os.RemoveAll(graphDir)
+	if err := d.StartWithBusybox("-g", graphDir); err != nil {
+		t.Fatal(err)
+	}
+
+	tmpDir := filepath.Join(os.TempDir(), "test")
+	defer os.RemoveAll(tmpDir)
+
+	if out, err := d.Cmd("create", "-v", fmt.Sprintf("%s:/foo", tmpDir), "--name=test", "busybox"); err != nil {
```

Comment:
@cpuguy83 no need for Sprintf if you only have one variable: `tmpDir + ":/foo"` is sufficient.